examples/kgcn/model.py

```diff
@@ -0,0 +1,102 @@
+import torch
+from aggregator import Aggregator
+import gs
+
+
+class KGCN(torch.nn.Module):
+    def __init__(self, num_user, num_ent, num_rel, kg: gs.Matrix, args, device):
+        super(KGCN, self).__init__()
+        self.num_user = num_user
+        self.num_ent = num_ent
+        self.num_rel = num_rel
+        self.n_iter = args.n_iter
+        self.batch_size = args.batch_size
+        self.dim = args.dim
+        self.n_neighbor = args.neighbor_sample_size
+        self.kg = kg
+        self.device = device
+        self.aggregator = Aggregator(
+            self.batch_size, self.dim, args.aggregator)
+
+        self._gen_adj()
+
+        self.usr = torch.nn.Embedding(num_user, args.dim)
+        self.ent = torch.nn.Embedding(num_ent, args.dim)
+        self.rel = torch.nn.Embedding(num_rel, args.dim)
+
+    def _gen_adj(self):
+        '''
+        Generate adjacency matrix for entities and relations
+        Only cares about fixed number of samples
+        '''
+        sampled_adj_matrix = self.kg.columnwise_sampling(self.n_neighbor, True)
+        self.adj_ent = sampled_adj_matrix.row_indices().view((self.num_ent, self.n_neighbor))
```

Comment:
An error may occur here because we don't do nothing to the node where deg=0, even to create memory space.

Reply:
Knowledge graph might not have this problem? I didn't handle this case here since I didn't see some special treatment in the author's version.

Reply:
Make sure it doesn't happen